### ISSUE TITLE
Change private icons and text to use lighter and less saturated color rather than transparency

### ISSUE THREAD

**Issue by jeisenbe** [gravitystorm/openstreetmap-carto#4078]
Related to #3589

Currently private icons use transparency (opacity 0.33) for the icon and text in several cases.

This started with `amenity=parking` + `access=private` in 2014 - see #1012, then private recycling and playgrounds were added in #1191.

But most icons and text labels with this transparent rendering were added in later 2018: #3528

Since these newer features can be rendered on various backgrounds, or overlapping two different colors, the low opacity (high transparency) leads to odd color mixing sometimes. The semi-transparent text labels can be hard to read (as mentioned in issue #3320 and #2170). 

There were also some suggestions to not render some of these features when they are private, e.g. #683.

I would suggest reconsidering this rendering. While it often works, when the icon is on a contrasting color (e.g. green on red, purple on yellow) the low opacity produces non-intuitive colors.

An alternative would be to reduce the saturation and increase the lightness for the private icons, rather than using `opacity`. This will make the colors the same no matter what background color is in the area. 

See https://github.com/gravitystorm/openstreetmap-carto/issues/3589#issuecomment-449633248 and https://github.com/gravitystorm/openstreetmap-carto/issues/3589#issuecomment-452949479

Current rendering:

https://tile.openstreetmap.org/19/274268/181238.png

![screenshot_2018-12-23 181238 png obraz png 256 x 256 pikseli](https://user-images.githubusercontent.com/5439713/50379517-8a0e0800-064b-11e9-87dc-dc3de6031fc0.png)

Test with current rendering over different color areas:

![opacity-test-current-master](https://user-images.githubusercontent.com/42757252/50978004-cc35b580-1536-11e9-9e3c-e885870646bb.png)

**Comment by polarbearing** [comment]
Yes I am in favour of having two fixed colours for public and private, respectively, instead of transparency. 
Question is if we simply define two values for a particular type, or want that calculate as a percentage?

**Comment by jeisenbe** [comment]
It would give the best results if we calculated the new color using Lch or Lab or another perceptually uniform colorspace. 

But it would be easier to use the lighten and desaturate functions in Mapnik. Those would also have the advantage of clearly relating the new color to the original color, and would make adapting the style easier in the future.

Another option is to mix the icon color with a neutral light gray color. See https://github.com/gravitystorm/openstreetmap-carto/issues/3589#issuecomment-452949479
